### ISSUE TITLE
A fatal bug!

### ISSUE THREAD

**Issue by Cluoyao** [harveyslash/Facial-Similarity-with-Siamese-Networks-in-Pytorch#27]
@Fanshia Hi, Fanshia, i found a question in your code, it's about contrastive loss function.
 `euclidean_distance = F.pairwise_distance(output1, output2)`

in PyTorch Docs
`def pairwise_distance(x1, x2, p=2., eps=1e-6, keepdim=False):`
[https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#pairwise_distance](url)

so,  "euclidean_distance" maybe it should be like this   `euclidean_distance = F.pairwise_distance(output1, output2, keepdim = True)`
that's why the loss curve is instability, and test performance is bad, If you modify it like this, you will found the performance is well! To be honest, this problem has also plagued me for a long time.  But anyway, thank you for your works, it's a very good tutorial.

**Comment by eldarovich99** [comment]
Thank you a lot, man!

**Comment by harveyslash** [comment]
A PR to rectify this error is most welcome!
